### ISSUE TITLE
Deployment fails and defaults to Amazon Linux 2023 instead of Amazon Linux 2

### ISSUE THREAD

**Issue by eggboy0** [zodern/mup-aws-beanstalk#180]
There are issues with deploying to Amazon Linux 2023 which mup-aws-beanstalk defaults to since Amazon Linux 2023's inception. How do we make it choose Amazon Linux 2 or does this plugin require a patch update to resolve the issue?

**Comment by zodern** [comment]
What are the problems with Amazon Linux 2023?

I'm guessing it is related to this code which selects the platform: https://github.com/zodern/mup-aws-beanstalk/blob/e5cb4137df4fd98f9ab032df035f5c671e895fe2/src/utils.js#L168-L213

**Comment by lynchem** [comment]
I believe we're encountering this as well now. We had our test server deactivated for a short time and when I went to deploy now it picked this new version of the platform as well and fails. 

Would it be possible to specify the platform branch somewhere in the config ? 

Edit: [Here's the changelog](https://docs.aws.amazon.com/linux/al2023/ug/compare-with-al2.html). Nothing jumps out at me. I'm seeing the health check fail.

**Comment by eggboy0** [comment]
> What are the problems with Amazon Linux 2023?
> 
> I'm guessing it is related to this code which selects the platform:
> 
> https://github.com/zodern/mup-aws-beanstalk/blob/e5cb4137df4fd98f9ab032df035f5c671e895fe2/src/utils.js#L168-L213

Yeah that's right. It seems like it was just picking the first one from the list.

I unfortunately didn't save the logs when the issue happened - I already deleted the environment.

As lynchem suggested, specifying the platform branch in the config would be nice.

**Comment by lynchem** [comment]
We since got ours all running fine. The new platform has a larger memory footprint so in one case we had to go to the next instance size to fix it.
